numpy/typing.pyi

```diff
@@ -15,6 +15,22 @@ _ShapeLike = Union[int, Sequence[int]]
 
 _DtypeLikeNested = Any  # TODO: wait for support for recursive types
 
+# Mandatory keys
+class _DtypeDictBase(TypedDict):
+    names: Sequence[str]
+    formats: Sequence[_DtypeLikeNested]
+
+# Mandatory + optional keys
+class _DtypeDict(_DtypeDictBase, total=False):
+    offsets: Sequence[int]
+    titles: Sequence[Union[bytes, Text, None]]
```

Comment:
```suggestion
    titles: Sequence[Any]
```
`np.dtype(dict(names=['a'], formats=[int], titles=[object()]))` is apparently allowed.

Reply:
Good call; fixed in https://github.com/numpy/numpy/pull/16622/commits/a96f1f1ab740c2509720df98c535f0034ed453de.